### ISSUE TITLE
Accessibility support

### ISSUE THREAD

**Issue by tomivirkki** [vaadin/vaadin-grid#1015]
Finish accessibility support

**Comment by platosha** [comment]
### Accessibility Feature Support Matrix

The following would have effect after PR #1023 is merged.

| OS + Reader + Browser | Grid and cell roles<br> (`role="grid"`, `role="gridcell"`) | Column headers announced on body cells<br> (`role="columnheader"`) | Total number of rows and columns<br> (`aria-rowcount`, `aria-colcount`) | Row numbers<br> (`aria-rowindex`) | Row selection<br> (`aria-selected`) | Row details toggle state<br> (`aria-expanded`) | Sorter state<br> (`aria-sort`) |
| :--- | :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| Windows 10 + NVDA 2017.3 + Chrome 61 | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :o:<br> Missing | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :warning:<br> Not live |
| Windows 10 + Jaws + Chrome 61 | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :o:<br> Missing | :x:<br> Invalid | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :warning:<br> Not live |
| Windows 10 + Narrator + Edge 16 | :white_check_mark:<br> Valid | :o:<br> Missing | :x:<br> Invalid | :x:<br> Invalid | :warning:<br> Not live | :white_check_mark:<br> Valid | :warning:<br> Not live |
| MacOS Sierra + VoiceOver + Safari 11.0 | :white_check_mark:<br> Valid | :o:<br> Missing | :x:<br> Invalid | :o:<br> Missing | :o:<br> Missing | :white_check_mark:<br> Valid | :o:<br> Missing |
| MacOS Sierra + VoiceOver + Chrome 61| :x:<br> Invalid | :o:<br> Missing | :x:<br> Invalid | :o:<br> Missing | :o:<br> Missing | :white_check_mark:<br> Valid | :o:<br> Missing |
| MacOS High Sierra + VoiceOver + Safari 11.0 | :white_check_mark:<br> Valid | :o:<br> Missing | :x:<br> Invalid | :o:<br> Missing | :o:<br> Missing | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid |
| MacOS High Sierra + VoiceOver + Chrome 61| :x:<br> Invalid | :o:<br> Missing | :x:<br> Invalid | :o:<br> Missing | :o:<br> Missing | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid |
| iOS 10.3 + VoiceOver + Safari | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :white_check_mark:<br> Valid | :o:<br> Missing | :o:<br> Missing | :o:<br> Missing |

TODO checks on:
- Android TalkBack
- iOS 11.

**Comment by knoobie** [comment]
Hey @platosha where did you find this support matrix or did you created it yourself?

**Comment by platosha** [comment]
@knoobie created myself after trying `<vaadin-grid>` with these platform/UA/reader combinations. Tried the PR branch back then, now this is aleady merged.

**Comment by knoobie** [comment]
@platosha Oh, thanks! Your self-study reflects the things I noticed with our sightless user - aria-sort in on Windows broken..
